benchmark/httpapi.py

```diff
@@ -277,18 +433,23 @@ def main(reactor, args):
         sys.stderr.write(options.getUsage())
         raise SystemExit(1)
 
+    def get_backend_from_options(options):
+        backend = _BACKENDS[options['backend']]
+
+        conn = dict()
+        if options['db-hostname']:
+            conn['hostname'] = options['db-hostname']
+        if options['db-port']:
+            conn['port'] = options['db-port']
+
+        return backend(**conn)
+
     startLogging(sys.stderr)
-    service = create_api_service(TCP4ServerEndpoint(reactor, options['port']))
 
-    # XXX Make startService() raise an exception on an error
-    # instead of just logging and dropping it.
```

Comment:
:question: I realise this comment isn't added by you, but it's not really clear to me
how we fix this. We don't seem to override that method in this codebase,
so is it referring to a change in Twisted.

If it's our code then moving it to the code that we should change would be
better. If it's twisted code then mentioning that, maybe with a bug reference
would be ideal.

If you aren't sure either then that's fine. If no-one is sure maybe we should
just delete the comment.